x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/settings_flyout/alert_selection/use_data_view/index.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { DataView, DataViewSpec } from '@kbn/data-views-plugin/common';
+import { useEffect, useState } from 'react';
+
+import { useKibana } from '../../../../../common/lib/kibana';
+
+export const useDataView = ({
+  dataViewSpec,
+  loading,
+}: {
+  dataViewSpec: DataViewSpec | undefined;
+  loading: boolean;
+}): DataView | undefined => {
+  const { dataViews } = useKibana().services;
+
+  const [dataView, setDataView] = useState<DataView | undefined>(undefined);
+
+  useEffect(() => {
+    let active = true;
+
+    async function createDataView() {
+      if (dataViewSpec != null && !loading) {
```

Comment:
This additional check was added because in the `8.x` branch, the `dataViewSpec` argument may be `undefined`: 

```typescript
dataViewSpec: DataViewSpec | undefined;
```

as noted in [this comment](https://github.com/elastic/kibana/pull/205137/files#r1896747892) above.